app/controllers/albums_controller.rb

```diff
@@ -0,0 +1,52 @@
+class AlbumsController < ApplicationController
+
+  def index
+    @albums = Album.all
+  end
+
+  def new
+    @album = Album.new
+  end
+
+  def create
+    @album = Album.new(album_params)
+    if @album.save
+      redirect_to action: 'index'
+    else
+      render :new
+    end
+  end
+
+  def show
+    @album = get_album
+  end
+
+  def edit
+    @album = get_album
+  end
+
+  def update
+    @album = Album.find(params[:id])
+    if @album.update_attributes(album_params)
+      redirect_to action: 'index'
+    else
+      render :edit
+    end
+  end
+
+    def destroy
+      Album.find_by(id: params['id']).destroy
+      redirect_to root_path
+    end
+
+  private
+
+  def album_params
+    params.require(:album).permit(:title, :artist, :genre)
+  end
+
+  def get_album
+    @album = Album.find_by(id: params['id'])
```

Comment:
You are already setting `@album = get_album`, so no need to assign the variable here. Just this:
```ruby
def get_album
  Album.find_by(id: params[:id])
end
```
Note: `params` are a `HashWithIndifferentAccess`, so you can use string or symbol to access that hash